src/EnergyPlus/ReportSizingManager.cc

```diff
@@ -529,11 +527,27 @@ namespace ReportSizingManager {
                                     if (FinalZoneSizing(CurZoneEqNum).CoolDDNum > 0 &&
                                         FinalZoneSizing(CurZoneEqNum).CoolDDNum <= DataEnvironment::TotDesDays) {
                                         DDNameFanPeak = WeatherManager::DesDayInput(FinalZoneSizing(CurZoneEqNum).CoolDDNum).Title;
+                                        dateTimeFanPeak =
+                                            General::TrimSigDigits(
+                                                WeatherManager::DesDayInput(FinalZoneSizing(CurZoneEqNum).CoolDDNum).Month) +
+                                            "/" +
+                                            General::TrimSigDigits(
+                                                WeatherManager::DesDayInput(FinalZoneSizing(CurZoneEqNum).CoolDDNum).DayOfMonth) +
+                                            " " +
+                                            coilSelectionReportObj->getTimeText(FinalZoneSizing(CurZoneEqNum).TimeStepNumAtCoolMax);
```

Comment:
Generally, code that follows a DRY concept is appreciated.  This code is like drenching wet I guess then?  I know you didn't add all these blocks that needed to be updated, so your changes make sense, but it just seems crazy that there are so many.

Reply:
Oh yeah it’s pretty wet alright!

I initially started to try to reduce it down but realized that save for changing these two lines with a flag I couldn’t do much in a short time. Reducing the if/else structure is really what would help, but the thought of having to create a flow diagram of all these cases to get a clear picture so I try to could boil it down was a bit discouraging (and the fan section I touched isn’t the only one, there are coils etc too) so I wrote a short python script to add the missing lines instead...

I think it could be another “project” (An enhancement rather than a bug) to clean *and* document this (a reduced flow diagram could help anyone having to touch this section down the line). Or I could do it in this one but that extends the scope seriously.

Reply:
No worries for now, this is just fine.  Thanks!

Reply:
@jmarrec Just to fill in the history, this was the first generation consolidation of sizing calculations which were duplicated in all of the components. Rewriting this has been discussed, but hasn't made it to the top of the list yet. For this, I know it's merged already, but it looks like there are only two variations on this. Could you just set two strings at the top and assign either the heat or cool version in each block with a single line?